File: sdk/ruby/lib/rocksky/resources/album.rb
module Rocksky
  module Resources
    # `app.rocksky.album.*` endpoints.
    class Album < Base
      # Fetch an album by AT-URI.
      def get_album(uri:)
        query("app.rocksky.album.getAlbum", uri: uri)
      end

      # List albums.
      def get_albums(limit: nil, offset: nil, genre: nil)
        query("app.rocksky.album.getAlbums",
              limit: limit, offset: offset, genre: genre)
      end

      # Tracks belonging to an album.
      def get_album_tracks(uri:)
        query("app.rocksky.album.getAlbumTracks", uri: uri)
      end
    end
  end
end

File: sdk/python/src/rocksky/resources/album.py
"""``app.rocksky.album.*`` — album views."""

from __future__ import annotations

from ..models import Album, AlbumBasic, SongBasic
from ._base import Resource, parse_list, parse_model


class AlbumResource(Resource):
    async def get(self, uri: str) -> Album:
        """Get a detailed album view by AT-URI."""
        data = await self._transport.query(
            "app.rocksky.album.getAlbum",
            params={"uri": uri},
        )
        return parse_model(Album, data)

    async def list(
        self,
        *,
        limit: int | None = None,
        offset: int | None = None,
        genre: str | None = None,
    ) -> list[AlbumBasic]:
        data = await self._transport.query(
            "app.rocksky.album.getAlbums",
            params={"limit": limit, "offset": offset, "genre": genre},
        )
        return parse_list(AlbumBasic, data, key="albums")

    async def get_tracks(self, uri: str) -> list[SongBasic]:
        data = await self._transport.query(
            "app.rocksky.album.getAlbumTracks",
            params={"uri": uri},
        )
        return parse_list(SongBasic, data, key="tracks")
